File: v1/examples/csharp/11_Emergency.cs
// 11_Emergency.cs - v1 API Emergency Call Device List Example
//
// Endpoint:
//   GET /api/v1/emergency  - Emergency call device list
//
// Note: Requires Emergency Call license. Returns 404 if not supported.

using System;
using System.Threading.Tasks;

namespace TsApiExamples.V1
{
    class Emergency
    {
        static async Task Main(string[] args)
        {
            using var client = new NvrClient();
            if (string.IsNullOrEmpty(client.ApiKey))
            {
                Console.WriteLine("NVR_API_KEY environment variable is required");
                return;
            }
            client.SetApiKey(client.ApiKey);

            // ── Emergency call device list ──
            Console.WriteLine("=== Emergency Call Devices ===");
            var r = await client.GetAsync("/api/v1/emergency");

            if ((int)r.StatusCode == 404)
            {
                Console.WriteLine("  Emergency Call not enabled on this server (license required)");
                return;
            }

            r.EnsureSuccessStatusCode();
            using var doc = await NvrClient.ReadJsonAsync(r);
            foreach (var dev in doc.RootElement.EnumerateArray())
            {
                var id = dev.GetProperty("id").GetInt32();
                var code = dev.GetProperty("code").GetString();
                var name = dev.GetProperty("name").GetString();
                var channels = dev.GetProperty("linkedChannel");
                Console.WriteLine($"  id={id}  code={code}  name={name}  linkedChannel={channels}");
            }
        }
    }
}

File: v1/examples/python/11_emergency.py
"""
11_emergency.py - TS-API v1 Emergency Call Device List

Endpoint:
  GET /api/v1/emergency  - Emergency call device list

Response:
  [
    {
      "id": 1,
      "code": "EM-001",
      "name": "Fire Alarm",
      "linkedChannel": [1, 2, 3]
    }
  ]

Note: Requires Emergency Call license. Returns 404 if not supported.
"""

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from config import NVR_API_KEY
from http_client import NvrClient

if not NVR_API_KEY:
    print('NVR_API_KEY environment variable is required')
    sys.exit(1)


def main():
    client = NvrClient()
    client.set_api_key(NVR_API_KEY)

    # ── Emergency Call Device List ──
    r = client.get('/api/v1/emergency')
    print(f'=== Emergency Devices === status={r.status_code}')

    if r.status_code == 200:
        devices = r.json()
        print(f'  Total: {len(devices)} device(s)')
        for dev in devices:
            chans = dev.get('linkedChannel', [])
            print(f'  id={dev["id"]}  code={dev.get("code", "")}'
                  f'  name={dev.get("name", "")}'
                  f'  linkedChannel={chans}')
    elif r.status_code == 404:
        print('  Emergency Call not enabled on this server (license required)')


if __name__ == '__main__':
    main()
